hack/test-extended/default.sh

```diff
@@ -0,0 +1,224 @@
+#!/bin/bash
+#
+# This scripts starts the OpenShift server with a default configuration.
+# The OpenShift Docker registry and router are installed.
+# It will start all 'default_*_test.go' test cases.
+
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
+
+source ${OS_ROOT}/hack/util.sh
+source ${OS_ROOT}/hack/common.sh
+
+echo "[INFO] Starting extended tests"
+
+TIME_SEC=1000
+TIME_MIN=$((60 * $TIME_SEC))
+
+TMPDIR="${TMPDIR:-"/tmp"}"
+BASETMPDIR="${TMPDIR}/openshift-extended-tests"
+
+# Use either the latest release built images, or latest.
+if [[ -z "${USE_IMAGES-}" ]]; then
+	USE_IMAGES='openshift/origin-${component}:latest'
+	if [[ -e "${OS_ROOT}/_output/local/releases/.commit" ]]; then
+		COMMIT="$(cat "${OS_ROOT}/_output/local/releases/.commit")"
+		USE_IMAGES="openshift/origin-\${component}:${COMMIT}"
+	fi
+fi
+
+
+if [[ -z "${BASETMPDIR-}" ]]; then
+	remove_tmp_dir && mkdir -p "${BASETMPDIR}"
+fi
+
+OS_TEST_NAMESPACE="extended-tests"
+
+ETCD_DATA_DIR="${BASETMPDIR}/etcd"
+VOLUME_DIR="${BASETMPDIR}/volumes"
+FAKE_HOME_DIR="${BASETMPDIR}/openshift.local.home"
+LOG_DIR="${LOG_DIR:-${BASETMPDIR}/logs}"
+ARTIFACT_DIR="${ARTIFACT_DIR:-${BASETMPDIR}/artifacts}"
+mkdir -p $LOG_DIR
+mkdir -p $ARTIFACT_DIR
+
+DEFAULT_SERVER_IP=`ifconfig | grep -Ev "(127.0.0.1|172.17.42.1)" | grep "inet " | head -n 1 | sed 's/adr://' | awk '{print $2}'`
+API_HOST="${API_HOST:-${DEFAULT_SERVER_IP}}"
+API_PORT="${API_PORT:-8443}"
+API_SCHEME="${API_SCHEME:-https}"
+MASTER_ADDR="${API_SCHEME}://${API_HOST}:${API_PORT}"
+PUBLIC_MASTER_HOST="${PUBLIC_MASTER_HOST:-${API_HOST}}"
+KUBELET_SCHEME="${KUBELET_SCHEME:-https}"
+KUBELET_HOST="${KUBELET_HOST:-127.0.0.1}"
+KUBELET_PORT="${KUBELET_PORT:-10250}"
+
+SERVER_CONFIG_DIR="${BASETMPDIR}/openshift.local.config"
+MASTER_CONFIG_DIR="${SERVER_CONFIG_DIR}/master"
+NODE_CONFIG_DIR="${SERVER_CONFIG_DIR}/node-${KUBELET_HOST}"
+
+# use the docker bridge ip address until there is a good way to get the auto-selected address from master
+# this address is considered stable
+# used as a resolve IP to test routing
+CONTAINER_ACCESSIBLE_API_HOST="${CONTAINER_ACCESSIBLE_API_HOST:-172.17.42.1}"
+
+STI_CONFIG_FILE="${LOG_DIR}/stiAppConfig.json"
+DOCKER_CONFIG_FILE="${LOG_DIR}/dockerAppConfig.json"
+CUSTOM_CONFIG_FILE="${LOG_DIR}/customAppConfig.json"
+GO_OUT="${OS_ROOT}/_output/local/go/bin"
+
+# set path so OpenShift is available
+export PATH="${GO_OUT}:${PATH}"
+
+cleanup() {
```

Comment:
If you take a closer look you will see that the methods are not so familiar. Al the `cleanup()` in the ext-tests is doing is stops OS server, stops and removes containers, removes the testing dir (+ unmounts any mounts used)